plugins/speculation-rules/settings.php

```diff
@@ -280,11 +307,63 @@ function plsr_render_settings_field( array $args ): void {
 			</p>
 		<?php endforeach; ?>
 
+		<?php if ( $show_notice ) : ?>
+			<div id="plsr-auth-notice" class="notice <?php echo esc_attr( $show_warning ? 'notice-warning' : 'notice-info' ); ?> inline">
+				<p>
+					<?php
+					echo wp_kses(
+						sprintf(
+							/* translators: %s: URL to persistent object cache documentation */
+							__( 'Enabling speculative loading for authenticated users without a persistent object cache may significantly increase server load. Consider setting up a <a href="%s" target="_blank">persistent object cache</a> before enabling this feature for logged-in users.', 'speculation-rules' ),
+							'https://developer.wordpress.org/advanced-administration/performance/optimization/#object-caching'
+						),
+						array(
+							'a' => array(
+								'href'   => array(),
+								'target' => array(),
+							),
+						)
+					);
+					?>
+				</p>
+			</div>
+		<?php endif; ?>
+
 		<p class="description" style="max-width: 800px;">
-			<?php echo esc_html( $args['description'] ); ?>
+			<?php
+			echo wp_kses(
+				$args['description'],
+				array(
+					'a' => array(
+						'href'   => array(),
+						'target' => array(),
+					),
+				)
+			);
+			?>
 		</p>
 	</fieldset>
+
 	<?php
+	if ( $show_notice ) {
+		// TODO: Unsure why this is not allowed in PHPCS since not mentioned in WP coding standards and heredocs/nowdocs are use in core: <https://github.com/search?q=repo%3AWordPress%2Fwordpress-develop+%2F%3C%3C%3C%27%3F%5BA-Z%5D%2B%2F&type=code>.
+		// phpcs:ignore Squiz.PHP.Heredoc.NotAllowed
```

Comment:
@b1ink0 Could you look into this? I tried quickly to see which PHPCS ruleset included this, but nothing jumped out at me. If we can identify this, then we can remove these lines:

```suggestion
```

Reply:
Oh, I found the issue. It is unrelated to our PHPCS ruleset. The cause is the ruleset inside the `wpackagist-plugin`. Look at the screenshot below: when I comment it out, the PHPCS error goes away and it passes the pre-commit checks. As this is in `vendor`, I am not sure why it is affecting outside of it.

<img width="1230" height="307" alt="Screenshot 2025-08-22 at 9 59 56 PM" src="https://github.com/user-attachments/assets/31c0eef1-e85e-4acf-9ce5-b628b8415395" />

Reply:
I ran this command in repo:
```bash
grep -r "Heredoc" . --include="*.xml" --
include="*.dist"
```
Output:
```log
./vendor/wp-coding-standards/wpcs/WordPress-Extra/ruleset.xml:    <rule ref="Generic.Strings.UnnecessaryHeredoc"/>
./vendor/wpackagist-plugin/plugin-check/phpcs-rulesets/plugin-check.ruleset.xml:  <rule ref="Squiz.PHP.Heredoc">
./vendor/wpackagist-plugin/plugin-check/vendor/squizlabs/php_codesniffer/src/Standards/Generic/Docs/Strings/UnnecessaryHeredocStandard.xml:<documentation title="Unnecessary Heredoc">
./vendor/wpackagist-plugin/plugin-check/vendor/squizlabs/php_codesniffer/src/Standards/Generic/Docs/WhiteSpace/HereNowdocIdentifierSpacingStandard.xml:<documentation title="Heredoc Nowdoc Identifier Spacing">
./vendor/wpackagist-plugin/plugin-check/vendor/squizlabs/php_codesniffer/src/Standards/Squiz/Docs/PHP/HeredocStandard.xml:<documentation title="Heredoc">
./vendor/squizlabs/php_codesniffer/src/Standards/Generic/Docs/Strings/UnnecessaryHeredocStandard.xml:<documentation title="Unnecessary Heredoc">
./vendor/squizlabs/php_codesniffer/src/Standards/Generic/Docs/WhiteSpace/HereNowdocIdentifierSpacingStandard.xml:<documentation title="Heredoc Nowdoc Identifier Spacing">
./vendor/squizlabs/php_codesniffer/src/Standards/Squiz/Docs/PHP/HeredocStandard.xml:<documentation title="Heredoc">
```

Reply:
Good find! I see it was introduced in https://github.com/WordPress/plugin-check/pull/800 to fix https://github.com/WordPress/plugin-check/issues/792.

I've appealed in https://github.com/WordPress/plugin-check/issues/792#issuecomment-3214985527.

For now, I think we should just proceed with suppressing the sniff:

```suggestion
		// phpcs:ignore Squiz.PHP.Heredoc.NotAllowed -- Part of the PCP ruleset. Appealed in <https://github.com/WordPress/plugin-check/issues/792#issuecomment-3214985527>.
```

Or we could exclude it in our ruleset.

Reply:
Okay, it's our repo's issue look at
https://github.com/WordPress/performance/blob/3d88f7b7c1ba643e0e85f310a4e0c922a4eed045/tools/phpcs/phpcs.ruleset.xml#L6

cc: @westonruter

Reply:
Commit responsible: https://github.com/WordPress/performance/commit/e16335b5fef867990e70f8d0f64d22ec9153674c

Reply:
We could change the severity of `Squiz.PHP.Heredoc.NotAllowed` to be zero in our ruleset, if we don't exclude it entirely.

Reply:
>Or we could exclude it in our ruleset.

Oh, I didn’t see your messages when I added mine, so I ended up commenting the same thing.

I think we should ignore it for now we can revert our change in the future.

Reply:
Please commit what you think is correct. I will be away for a few hours.